packages/react-core/src/components/Truncate/Truncate.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Truncate/truncate';
+import { css } from '@patternfly/react-styles';
+import { Tooltip } from '../Tooltip';
+
+export enum TruncatePosition {
+  start = 'start',
+  end = 'end',
+  middle = 'middle'
+}
+
+const truncateStyles = {
+  start: styles.truncateEnd,
+  end: styles.truncateStart
+};
+
+interface TruncateProps extends React.HTMLProps<HTMLDivElement> {
+  /** Class to add to outer span */
+  className?: string;
+  /** Text to truncate */
+  content: string;
```

Comment:
I think that if we know that the content will always be a string, there is no need to have it as children and just use a prop content. But this was one thing that I knew that someone will be commenting, so totally up for debate :D

Reply:
I wonder if at some point we'd want `content` to be bold or italicized. In that case, I think we'd want to just use ReactNode. If there's potential for that, I think it's worth a debate, or at least the changing of the type, otherwise if we know it will always be a string, what you have is fine with me.

Reply:
This is a good question for @mattnolting 
Will the css that controls the truncation still work if someone wants to put an icon or link somewhere in the string?

Reply:
The text can be bold or italicized, you can just add those tags around. But with the icon it won't work. So I think that it is best we keep it as string.